app/src/main/java/com/mithridat/nonoconverter/ui/editimage/EditImageActivity.java

```diff
@@ -467,4 +552,65 @@ void unLink() {
             _activity = null;
         }
     }
-}
+
+    /**
+     * Close the given closeable object (Stream) in a safe way:
+     * check if it is null and catch.
+     *
+     * @param closeable the closable object to close
+     */
+    private static void closeSafe(Closeable closeable) {
+        if (closeable != null) {
+            try {
+                closeable.close();
+            } catch (IOException ignored) {}
+        }
+    }
+
+    /**
+     * Write given bitmap to a temp file. If file already exists no-op as we
+     * already saved the file in this session. Uses JPEG 100% compression.
+     *
+     * @param uri the uri to write the bitmap to, if null
+     * @return the uri where the image was saved in, either the given uri or
+     * new pointing to temp file.
+     */
+    Uri writeTempStateStoreBitmap(Context context, Bitmap bitmap, Uri uri) {
+        try {
+            if(uri == null) {
+                uri =
+                        Uri.fromFile(
+                                File.createTempFile("ic_state_store_temp",
+                                        ".jpg",
+                                        context.getCacheDir()));
+            }
+            if(_needSaveCropped) {
+                writeBitmapToUri(
```

Comment:
Check `Long parameters case` in [Coding standard](https://github.com/nonoteam/NonoConverter/blob/master/CodingStandard.md#line-wrapping-strategies)